src/core/device.c

```diff
@@ -766,7 +766,30 @@ static int device_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 return 0;
         }
 
-        if (streq(action, "remove"))  {
+        if (streq(action, "change"))  {
+                _cleanup_free_ char *e = NULL;
+                Unit *u;
+
+                r = unit_name_from_path(sysfs, ".device", &e);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name from device path: %m");
+
+                u = manager_get_unit(m, e);
+                if (!u) {
+                        log_error("Failed to get unit from sysfs name.");
+                        return 0;
+                }
+
+                if (!unit_can_reload(u))
+                        return 0;
```

Comment:
this check is not really necessary, is it? manager_add_job() calls transaction_add_job_and_dependencies(), which calls unit_job_is_applicable(), which calls unit_can_reload(), hence we end up checking this twice. Let's instead just do this in manager_add_job